File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        vector<int> res(256,-1);
        int start=-1, len=0;
        for(int i=0;i<s.size();i++){
            if(res[s[i]]>start)
            start=res[s[i]];
            res[s[i]]=i;
            len=max(len,i-start);
        }
        return len;
    }
};

File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        res=[-1]*128
        start=-1
        siz=0
        for i in range(len(s)):
            if res[ord(s[i])-ord('a')]>start:
                start=res[ord(s[i])-ord('a')]
            res[ord(s[i])-ord('a')]=i
            siz=max(siz,i-start)
        return siz
        